src/main/kotlin/com/sourcegraph/cody/chat/ui/LlmDropdown.kt

```diff
@@ -67,59 +66,52 @@ class LlmDropdown(
   }
 
   @RequiresEdt
-  private fun updateModelsInUI(models: List<Model>) {
+  private fun updateModelsInUI(models: List<ModelAvailabilityStatus>) {
     if (project.isDisposed) return
     this.removeAllItems()
 
-    val availableModels = models.filterNot { it.isDeprecated() }
-    availableModels.sortedBy { it.isCodyProOnly() }.forEach(::addItem)
+    val availableModels = models.map { it }.filterNot { it.model.isDeprecated() }
+    availableModels.sortedBy { it.model.isCodyProOnly() }.forEach { addItem(it) }
 
     val selectedFromChatState = chatModelFromState
     val selectedFromHistory = HistoryService.getInstance(project).getDefaultLlm()
 
     selectedItem =
-        models.find {
-          it.id == model ||
-              it.id == selectedFromHistory?.model ||
-              it.id == selectedFromChatState?.id
-        } ?: models.firstOrNull()
-
-    val isEnterpriseAccount =
-        CodyAuthenticationManager.getInstance().account?.isEnterpriseAccount() ?: false
+        availableModels
+            .map { it.model }
+            .find {
+              it.id == model ||
+                  it.id == selectedFromHistory?.model ||
+                  it.id == selectedFromChatState?.id
+            } ?: models.firstOrNull()
 
     // If the dropdown is already disabled, don't change it. It can happen
     // in the case of the legacy commands (updateAfterFirstMessage happens before this call).
     isEnabled = isEnabled && chatModelFromState == null
-
-    isVisible = !isEnterpriseAccount || hasServerSentModels
```

Comment:
@jamesmcnamara After cody commit bump this seems to be not working anymore.

Could you remind me what was the rationale for `hasServerSentModels` introduction?
I think either way we should decide that on the Cody level, and if some conditions are not meet we should simply not sent models through this endpoint. This way we won't have custom, error-prone logic on the client side.

Also, looks like Cody is not allowing to choose between enterprise models.

Reply:
I think that these could be some leftovers related to chats (which are redundant now as we use the webview UI). You can doublecheck that Edit Code dialog works properly (this is the only place where we use the dropdown presently).

Reply:
I did: see the testplan :)